bitbots_body_behavior/src/bitbots_body_behavior/main.dsd

```diff
@@ -37,50 +37,84 @@ $BallSeen
     NO --> @Stop, @LookAtBall
 
 #Positioning
-$Localization
+$LocalizationAvailable
     YES --> @GoToRolePosition, @LookAtFieldFeatures
     NO --> @GoToRelativePosition + x:2 + y:0 + t:0
 
 #PositioningReady
-$Localization
+$LocalizationAvailable
     YES --> @GoToRolePosition, @LookAtFieldFeatures
     NO --> @GoToRelativePosition + x:2 + y:0 + t:0, @StandAndWait
 
 #GoalieBehavior
-$Localization
-    YES --> $JustUnpenalized
-        YES --> #Positioning
-        NO --> $BallSeen
-            YES --> $BallDangerous
-                YES --> #GoAndKickBallMapGoal
-                NO --> @Stop, @LookAtBall
+$LocalizationAvailable
+    YES --> $LocalizationPrecision
+        HIGH --> $TimerRunning + name:reorientation
+            YES --> @EndTimer + name:reorientation, @StartTimer + name:reorientation_pause + duration:%behavior/body/reorientation_pause_duration
+            NO --> $JustUnpenalized
+                YES --> #Positioning
+                NO --> $BallSeen
+                    YES --> $BallDangerous
+                        YES --> #GoAndKickBallMapGoal
+                        NO --> @Stop, @LookAtBall
+                    NO --> @SearchBall
+        LOW --> $TimerEnded + name:reorientation_pause
+            NO --> #NoLocalizationGoalieBehavior
+            YES --> $TimerRunning + name:reorientation
+                YES --> $TimerEnded + name:reorientation
+                    YES --> @StartTimer + name:reorientation_pause + duration:%behavior/body/reorientation_pause_duration
+                    NO --> @LookAtFieldFeatures, @Stop
+                NO --> @StartTimer + name:reorientation + duration:%behavior/body/reorientation_duration
     NO --> $JustUnpenalized
         YES --> @GoToRelativePosition + x:2 + y:0 + t:0
         NO --> #NoLocalizationGoalieBehavior
 
-#OffenseBehavior
-$Localization
-    YES --> $JustUnpenalized
-        YES --> #Positioning
-        NO --> $BallSeen
-            YES --> #GoAndKickBallMapGoal
-            NO --> @Stop, @SearchBall
-    NO --> $JustUnpenalized
+#NoLocalizationOffenseBehavior
+$JustUnpenalized
         YES --> @GoToRelativePosition + x:2 + y:0 + t:0
         NO --> #NoLocalizationPlayerBehavior
 
+#OffenseBehavior
+$LocalizationAvailable
```

Comment:
The part in the dad tree where the reorientation part is implemented seems a bit duplicated for the offernse and defense behavior. Is there any chance to reduce this duplication?

Reply:
Not to my knowledge. Because the behavior we return to after the similar part is different.